janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java

```diff
@@ -266,6 +266,13 @@
                     "performance improvement if there is a non-trivial latency to the backend.",
             ConfigOption.Type.MASKABLE, false);
 
+    public static final ConfigOption<Boolean> LIMIT_BATCH_SIZE = new ConfigOption<>(QUERY_NS,"limit-batch-size",
+            "Configure a maximum batch size for queries against the storage backend. This can be used to ensure " +
+            "responsiveness if batches tend to grow very large. The used batch size is equivalent to the " +
+            "barrier size of a preceding barrier() step. If a step has no preceding barrier(), the default barrier of TinkerPop " +
+            "will be inserted.",
```

Comment:
Might be worth noting that this option does not take effect if query.batch = false.

Only looking at the doc, I am not 100% sure how this could affect my use case. Can you maybe also explain in the doc what the default size of the TinkerPop barrier is? Does this apply to any place that MULTIQUERY is applicable, i.e. if this is enabled, all multi-queries will have an upper bound for batch size?

Reply:
Yes, all multi-queries of a query will be affected by this. If a user wishes to exclude certain multi-queries from the limit, they can prepend `barrier(Integer.MAX_VALUE)` for example.
I'm planning to write an entire docs page in the next few days to explain this feature.